### ISSUE TITLE
Do not wait for a DNSProvider to be Ready when a Shoot is hibernated

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/gardener-extension-shoot-dns-service#111]
/area quality
/kind bug

**What happened**:
With gardener-extension-shoot-dns-service@1.18.1 deployed on a landscape I see a lot of hibernated Shoots that fail to be reconciled with:
```yaml
  lastErrors:
    - description: "task \"Waiting until extension resources are ready\" failed: Error while waiting for Extension shoot--foo--bar/shoot-dns-service to become ready: error during reconciliation: Error reconciling extension: 1 error occurred:\n\t* Error while waiting for DNSProvider shoot--foo--bar/external to become ready: observed generation outdated (0/1)\n\n"
      taskID: Waiting until extension resources are ready
      lastUpdateTime: '2022-03-11T15:38:27Z'
```

The corresponding `Extension` resource:
```yaml
  lastError:
    description: "Error reconciling extension: 1 error occurred:\n\t* Error while
      waiting for DNSProvider shoot--foo--bar/external to become ready:
      observed generation outdated (0/1)\n\n"
    lastUpdateTime: "2022-03-11T15:41:59Z"
```

The corresponding dnsprovider does not seem to be reconciling because the Shoot is hibernated.

```
$ k -n shoot--foo--bar get dnsprovider
NAME       TYPE   STATUS   AGE   INCLUDED_DOMAINS
external                   97m
```

I assume the dns-controller-manager deployed as part of the control plane has to reconcile the dnsprovider but this cannot happen as the whole control plane scaled down.

**What you expected to happen**:
Reconciliation for a hibernated Shoot to succeed and the component to do not wait for a condition that cannot be completed.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a Shoot

2. Hibernate it.

3. Make sure reconciliation of the hibernated Shoot fails with the above error.

**Environment**:

- Gardener version (if relevant):
- Extension version: v1.18.1
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

**Comment by ialidzhikov** [comment]
cc @MartinWeindel

**Comment by MartinWeindel** [comment]
#112 is also included in v1.18.2
